### PR TITLE
feat(atlas-web-components): #183869844 migrate basic button to wc sta…

### PR DESCRIPTION
**Pull request by demogar**
# Feature

## Feature Description

- Change BasicButton implementation to be a Web Components wrapper
- Implemented Web Components package and project (built in Preact)
- Created the build process for development
- Changed the Turbo configuration to support new environment

## Screenshot

Screenshot or gif of the changes made.

## Ticket

[Ticket 183869844](https://www.pivotaltracker.com/story/show/183869844)

## Checklist

- [ ] Tests
- [x] Validated Against Design

### CODE REVIEW

File: packages/atlas-react/tests/components/BasicButton.test.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
-import { render, screen } from "@testing-library/react";
+import { render } from "@testing-library/react";
+import { screen } from "shadow-dom-testing-library";
```
Comment (demogar) [inline packages/atlas-react/tests/components/BasicButton.test.js:3]: @kvuzet looks like there is a way, by using [this library](https://www.npmjs.com/package/shadow-dom-testing-library), to use the shadow DOM that is generated by the WC, which is better than mocking the WC for tests.

Comment (nullcitizen) [reply packages/atlas-react/tests/components/BasicButton.test.js:3]: incredible!

File: packages/atlas-web-components/src/app.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useState } from 'preact/hooks'
+import preactLogo from './assets/preact.svg'
+import './components/BasicButton'
+import './app.css'
+
+export function App() {
+  return (
+    <>
+      <div>
+        <a href="https://vitejs.dev" target="_blank">
```
Comment (nullcitizen) [inline packages/atlas-web-components/src/app.jsx:10]: what's vite doing in here?

Comment (nullcitizen) [reply packages/atlas-web-components/src/app.jsx:10]: oh wait, I was thinking of lit

**Review by nullcitizen — APPROVED** [review]
hell yeah this looks amazing

**Pull request merged**